app/scaffold/pkgs/resolver.go

```diff
@@ -49,7 +51,24 @@ func (r *Resolver) Resolve(arg string, checkDirs []string) (path string, err err
 				Progress: os.Stdout,
 			})
 			if err != nil {
-				return "", fmt.Errorf("failed to clone repository: %w", err)
+				if err.Error() == "authentication required" {
+					username, password, err := promptHTTPAuth()
+					if err == nil {
+						r, err = git.PlainClone(dir, false, &git.CloneOptions{
+							URL:      remoteRef,
+							Progress: os.Stdout,
+							Auth: &http.BasicAuth{
+								Username: username,
+								Password: password,
+							},
+						})
+					}
```

Comment:
I must have deleted this comment on accident, whoops!

---

I'm wondering if this should be configuration driven in additon to being interactive? Interactive is nice, but if you're pulling down a few it would be nice to able to define these in the conf file

Here's what that could look like

```yaml
auth:
  - match: ^https://github.com.*
    basic:
      username: hay-kot 
      password: my-password
  - match: ^https://mylocalgit.com.*
    basic:
      username: ${MY_LOCAL_GIT_USERNAME}
      password: ${MY_LOCAL_GIT_PASSWORD}
```

Reply:
Oh very nice, yeah I think this would be great. Would you be alright with merging this as a first pass, and then adding config support as a fast follow? I'm happy to take a stab at the config work, too! But I'd also like to get this in :D

Reply:
Sounds good!